c-deps/libroach/engine.cc

```diff
@@ -302,4 +302,23 @@ DBStatus DBImpl::EnvDeleteFile(DBSlice path) {
   return ToDBStatus(status);
 }
 
+// EnvDeleteDir deletes the directory with the given dir name.
+DBStatus DBImpl::EnvDeleteDir(DBSlice dir) {
```

Comment:
Based on current testing, I didn't see subdirectories are created. If they are created, current approach will fail because when calling env's `DeleteDir`, it will return error such as `Directory not empty`.

Reply:
Except it won't, the `MemEnv` just deletes the entry, just like a file. You could delete the directory with stuff still in it (including files) and upon recreating it, the contents would magically re-appear. Yes, the MemEnv is pretty broken.

Reply:
Hmm, interesting, I will take a closer look.

Reply:
I'm not sure if it's relevant or not. But when I tried to call env's `DeleteDir` without deleting its children first, I got some `Directory not empty` error. That's also why I thought having subdirectories will throw `Directory not empty` error.